compat/src/index.d.ts

```diff
@@ -46,6 +46,7 @@ declare namespace React {
 	export import RefObject = preact.RefObject;
 	export import Component = preact.Component;
 	export import FunctionComponent = preact.FunctionComponent;
+	export type ComponentType<Props = {}> = Component<Props> | FunctionComponent<Props>;
```

Comment:
```suggestion
	export import preact.ComponentType;
```

We already have this type:

https://github.com/preactjs/preact/blob/4f2a04f5a6b0d043e577fbd2e76e4288e46ee089/src/index.d.ts#L78